### PR TITLE
Merge series/1.x to series/2.x

### PR DESCRIPTION
**Pull request by sideeffffect**
Please merge as _Merge_ commit (not Squash commit) :pray:

### CODE REVIEW

**Comment by sideeffffect** [comment]
@mijicd @adamgfraser Please merge as _Merge_ commit (not Squash commit) :pray:

**Review by adamgfraser — APPROVED** [review]

**Comment by sideeffffect** [comment]
Good to be merged @adamgfraser @mijicd :tada:

File: core/native/src/main/scala/zio/UpdateRuntimeFlagsWithinPlatformSpecific.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package zio
+
+// This is split because `java.util.function.IntFunction` isn't available in Scala Native yet.
+// Once it's available, this should be merged together with the `js-jvm` variant back to `shared`.
+// https://github.com/scala-native/scala-native/pull/3127
+private[zio] trait UpdateRuntimeFlagsWithinPlatformSpecific {
+  type RuntimeFlagsFunc[A] = Int => A
```
Comment (sideeffffect) [inline core/native/src/main/scala/zio/UpdateRuntimeFlagsWithinPlatformSpecific.scala:7]: btw, do we need to mark it as `private[zio]` too?

Comment (adamgfraser) [reply core/native/src/main/scala/zio/UpdateRuntimeFlagsWithinPlatformSpecific.scala:7]: I think if the trait is package private and the only object that extends it is package private there is no need for the members to be package private as well.

**Pull request merged**